ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/fillers/CoreProblemFiller.java

```diff
@@ -26,79 +27,163 @@ public CoreProblemFiller(LinearRaoProblem linearRaoProblem, LinearRaoData linear
-            crac.getCnecs().forEach(cnec -> {
-                fillCnec(cnec);
-                rangeActions.forEach(rangeAction -> updateCnecConstraintWithRangeAction(cnec, rangeAction));
-            });
-        }
+        // add variables
+        buildFlowVariables();
+        buildRangeActionSetPointVariables();
+        buildRangeActionAbsoluteVariationVariables();
+
+        // add constraints
+        buildFlowConstraints();
+        buildRangeActionConstraints();
     }
 
     @Override
-    public void update(List<String> activatedRangeActionIds) {
-        Crac crac = linearRaoData.getCrac();
-        if (crac.getPreventiveState() != null) {
-            Set<RangeAction> rangeActions = crac.getRangeActions(linearRaoData.getNetwork(), crac.getPreventiveState(), UsageMethod.AVAILABLE);
-            activatedRangeActionIds.forEach(this::updateRangeActionBounds);
-            crac.getCnecs().forEach(cnec -> {
-                linearRaoProblem.updateReferenceFlow(cnec.getId(), linearRaoData.getReferenceFlow(cnec));
-                rangeActions.forEach(rangeAction -> updateCnecConstraintWithRangeAction(cnec, rangeAction));
-            });
-        }
+    public void update() {
+        // update reference flow and sensitivities of flow constraints
+        updateFlowConstraints();
+    }
+
+    /**
+     * Build one flow variable F[c] for each Cnec c
+     * This variable describes the estimated flow on the given Cnec c, in MEGAWATT
+     */
+    private void buildFlowVariables() {
+        linearRaoData.getCrac().getCnecs().forEach(cnec ->
+                linearRaoProblem.addFlowVariable(-linearRaoProblem.infinity(), linearRaoProblem.infinity(), cnec)
+        );
     }
 
     /**
-     * Adds a cnec variable
+     * Build one set point variable S[r] for each RangeAction r
+     * This variable describes the set point of the given RangeAction r, given :
+     * <ul>
+     *     <li>in DEGREE for PST range actions</li>
+     * </ul>
+     *
+     * This set point of the a RangeAction is bounded between the min/max variations
+     * of the RangeAction :
      *
-     * @param cnec: cnec to add to optimisation problem
+     * initialSetPoint[r] - maxNegativeVariation[r] <= S[r]
+     * S[r] >= initialSetPoint[r] + maxPositiveVariation[r]
      */
-    private void fillCnec(Cnec cnec) {
-        linearRaoProblem.addCnec(cnec.getId(), linearRaoData.getReferenceFlow(cnec));
+    private void buildRangeActionSetPointVariables() {
+        linearRaoData.getCrac().getRangeActions().forEach(rangeAction -> {
+            double initialSetPoint = linearRaoData.getCurrentValue(rangeAction);
+            double maxNegVariation = rangeAction.getMaxNegativeVariation(linearRaoData.getNetwork());
+            double maxPosVariation = rangeAction.getMaxPositiveVariation(linearRaoData.getNetwork());
+            linearRaoProblem.addRangeActionSetPointVariable(initialSetPoint - maxNegVariation, initialSetPoint + maxPosVariation, rangeAction);
+        });
```

Comment:
So you can use directly getMinValue and getMaxValue

Reply:
Seems less twisted indeed